src/fhir-products/fhir-products.module.ts

```diff
@@ -0,0 +1,10 @@
+import { Module } from '@nestjs/common';
+import { FhirProductsController } from './fhir-products.controller';
+import { ProductsService } from 'src/products/products.service';
+import { OclClient } from 'src/lib/ocl/client';
+
+@Module({
+  controllers: [FhirProductsController],
+  providers: [ProductsService, OclClient],
```

Comment:
Since the FHIR layer is more of a formatting layer I think just delegating the resource fetching to the ProductService is better instead of adding the OclClient as a provider as well.

Reply:
the Oclclient is injected in the productService, as such it's erroring out without it. May be you can as well advise on how i can approach it.

Reply:
If it is working that is okay. Though convention is to export the provider from the products module then import the products module into the fhir module after which the service becomes visible to the fhir components.
[https://docs.nestjs.com/modules](More here.)

Reply:
aaaah thats great, will also do the refactor